Explorer/Assets/Scripts/CRDT/Deserializer/CRDTDeserializer.cs

```diff
@@ -139,22 +141,36 @@ public bool TryDeserializePutComponent(ref ReadOnlyMemory<byte> memory, out CRDT
 
             DeserializerParameters(ref memorySpan, out CRDTEntity entityId, out int componentId, out int timestamp, out int dataLength, ref shift);
 
-            if (TryReturnInvalidDataLength(ref memory, dataLength, memorySpan, shift))
-                return false;
+            try
+            {
+                if (TryReturnInvalidDataLength(ref memory, dataLength, memorySpan, shift))
+                    return false;
 
-            IMemoryOwner<byte> memoryOwner = crdtPooledMemoryAllocator.GetMemoryBuffer(memory, shift, dataLength);
+                IMemoryOwner<byte> memoryOwner = crdtPooledMemoryAllocator.GetMemoryBuffer(memory, shift, dataLength);
 
-            //Forwarding the memory to the next message
-            memory = memory.Slice(shift + dataLength);
+                //Forwarding the memory to the next message
+                memory = memory.Slice(shift + dataLength);
 
-            crdtMessage = new CRDTMessage(CRDTMessageType.PUT_COMPONENT, entityId, componentId, timestamp, memoryOwner);
-            return true;
+                crdtMessage = new CRDTMessage(CRDTMessageType.PUT_COMPONENT, entityId, componentId, timestamp, memoryOwner);
+                return true;
+            }
+            catch (Exception e)
```

Comment:
I would not wrap all the calls with `try...catch` as this area is very performance sensitive. Moreover, we can't recover from it. If we want an exception I would do like this:
- Check values manually and throw a meaningful custom exception
    - `Timestamp` should be dependent on `ISceneStateProvider.TickNumber`
    - `EntityId` and `Component` can never be negative
    - `DataLength` should be appropriate
- I would even change `TryReturnInvalidDataLength` to throw instead as we can't recover anyway
- I would not print the whole span as it's not going to give us anything meaningful. We rather need a history. But it's a completely different store